pairing/bn256/scalar.go

```diff
@@ -0,0 +1,148 @@
+package bn256
+
+import (
+	"crypto/cipher"
+	"errors"
+	"io"
+	"math/big"
+
+	"github.com/dedis/kyber"
+	"github.com/dedis/kyber/util/random"
+)
+
+type scalar struct {
+	x *big.Int
+}
+
+func NewScalar() kyber.Scalar {
+	s := &scalar{x: &big.Int{}}
+	return s.Zero()
+}
+
+func (s *scalar) Equal(a kyber.Scalar) bool {
+	return s.x.Cmp(a.(*scalar).x) == 0
+}
+
+func (s *scalar) Set(a kyber.Scalar) kyber.Scalar {
+	buf, _ := a.(*scalar).MarshalBinary()
+	s.UnmarshalBinary(buf)
+	return s
+}
+
+func (s *scalar) Clone() kyber.Scalar {
+	a := NewScalar()
+	a.Set(s)
+	return a
+}
+
+func (s *scalar) SetInt64(v int64) kyber.Scalar {
+	s.x.SetInt64(v)
+	return s
+}
+
+func (s *scalar) Zero() kyber.Scalar {
+	return s.SetInt64(0)
+}
+
+func (s *scalar) Add(a, b kyber.Scalar) kyber.Scalar {
+	ax := a.(*scalar).x
+	bx := b.(*scalar).x
+	s.x.Add(ax, bx)
+	s.x.Mod(s.x, Order)
+	return s
+}
+
+func (s *scalar) Sub(a, b kyber.Scalar) kyber.Scalar {
+	ax := a.(*scalar).x
+	bx := b.(*scalar).x
+	s.x.Sub(ax, bx)
+	s.x.Mod(s.x, Order)
+	return s
+}
+
+func (s *scalar) Neg(a kyber.Scalar) kyber.Scalar {
+	ax := a.(*scalar).x
+	s.x.Neg(ax)
+	s.x.Mod(s.x, Order)
+	return s
+}
+
+func (s *scalar) One() kyber.Scalar {
+	return s.SetInt64(1)
+}
+
+func (s *scalar) Mul(a, b kyber.Scalar) kyber.Scalar {
+	ax := a.(*scalar).x
+	bx := b.(*scalar).x
+	s.x.Mul(ax, bx)
+	s.x.Mod(s.x, Order)
+	return s
+}
+
+func (s *scalar) Div(a, b kyber.Scalar) kyber.Scalar {
+	ax := a.(*scalar).x
+	bx := b.(*scalar).x
+	s.x.Div(ax, bx)
+	s.x.Mod(s.x, Order)
+	return s
+}
+
+func (s *scalar) Inv(a kyber.Scalar) kyber.Scalar {
+	ax := a.(*scalar).x
+	s.x.ModInverse(ax, Order)
+	return s
+}
+
+func (s *scalar) Pick(rand cipher.Stream) kyber.Scalar {
+	buf := make([]byte, s.MarshalSize())
+	random.Bytes(buf, rand)
+	s.SetBytes(buf)
+	return s
+}
+
+func (s *scalar) SetBytes(buf []byte) kyber.Scalar {
+	s.UnmarshalBinary(buf)
+	return s
+}
+
+func (s *scalar) Bytes() []byte {
+	return s.x.Bytes()
+}
+
+func (s *scalar) SetVarTime(varTime bool) error {
```

Comment:
This is no longer needed/wanted. The new way to do what that was doing is: https://godoc.org/github.com/dedis/kyber#AllowsVarTime

But in your case, since this is a 100% vartime library, it should be marked "vartime" with build flags. (AllowsVarTime is for a predominantly constant time library to let callers opt in to vartime on a point by point basis.)